static/js/protein/data_processing_utils.ts

```diff
@@ -592,3 +537,319 @@ export function getProteinDescription(data: GraphNodes): string {
+  );
+  return _.get(scoreRec, `${protein1}_${protein2}`, defaultScore);
+}
+
+/**
+ * Given an object mapping interaction IDs to confidence scores and the DCIDs of an interaction, return the score
+ * of the interaction if it appears in the object, or the default score otherwise.
+ */
+export function scoreFromInteractionDCID(
+  scoreRec: Record<string, number>,
+  interactionDCID: string,
+  defaultScore: number = DEFAULT_INTERACTION_SCORE
+): number {
+  return _.get(scoreRec, idFromDCID(interactionDCID), defaultScore);
+}
+
+/**
+ * Given response and key, map each response datum to value of key and return map
+ */
+export function getFromResponse<
+  T extends BaseDCDataType,
+  K extends keyof V1ResponseDatum<T>
+>(resp: V1Response<T>, key: K): V1ResponseDatum<T>[K][] {
+  if (!("data" in resp.data)) return [];
+  return resp.data.data.map((obj) => obj[key]);
+}
+
+// Useful special cases of getFromResponse
+// Note that due to current limitations of TS indexed access types, "values" and "entity" need to be passed in twice -
+// once as a type and once as a string
+// Reference (last two paragraphs): https://www.typescriptlang.org/docs/handbook/2/indexed-access-types.html
+
+/**
+ * Given response, extract and return values (see def of V1Response)
+ */
+export function valuesFromResponse<T extends BaseDCDataType>(
+  resp: V1Response<T>
+): T[][] {
+  return getFromResponse<T, "values">(resp, "values");
+}
+
+/**
+ * Given response, extract and return DCIDs (entities)
+ */
+export function dcidsFromResponse<T extends BaseDCDataType>(
+  resp: V1Response<T>
+): string[] {
+  return getFromResponse<T, "entity">(resp, "entity");
+}
+
+/**
+ * Given an array of interaction DCIDs and a parallel array of corresponding scores,
+ * construct and return score record such that for each interaction A_B with corresponding DCID in interaction DCIDs,
+ * both A_B and B_A map to the score of A_B.
+ */
+export function symmetricScoreRec(
+  interactionDCIDs: string[],
+  scores: number[]
+): Record<string, number> {
+  const scoreRec = {};
+  zip(interactionDCIDs, scores).forEach(([dcid, score]) => {
+    const [protein1, protein2] = proteinsFromInteractionDCID(dcid);
+    scoreRec[`${protein1}_${protein2}`] = score;
+    scoreRec[`${protein2}_${protein1}`] = score;
+  });
+  return scoreRec;
+}
+
+/**
+ * Given score response, construct a record mapping interaction IDs of the form A_B to their confidence scores,
+ * satisfying the property that if A_B: A_B.score is in the record, then B_A: A_B.score is also.
+ */
+export function scoreDataFromResponse(
+  scoreResponse: V1Response<BaseDCDataType>
+): Record<string, number> {
+  const scoreValues = valuesFromResponse(scoreResponse);
+  const interactionDCIDs = dcidsFromResponse(scoreResponse);
+  const scoreList = scoreValues
+    // filter results for IntactMiScore - there should be 0 or 1 match
+    .map((scoreRecList: BaseDCDataType[] | undefined) => {
+      if (scoreRecList !== undefined) {
```

Comment:
Codacy doesn't like this check for if `scoreRecList` is `undefined`.  What's the proper way to do things?  Thanks!

Reply:
you could just check `!_.isEmpty(scoreRecList)`

Reply:
Thanks! I'll do this

Reply:
I'm sure you've experienced more weird null cases than I have in your lifetime, so I'm gonna add it!